File: DotNet/AuthService/src/AuthService.Domain/Base/BaseEntity.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AuthService.Domain.Base
{
    public abstract class BaseEntity
    {
        public Guid Id { get; set; }
        public DateTime CreatedAt { get; protected set; }
        public DateTime UpdatedAt { get; protected set; }
    }
}

File: Java/social/src/main/java/com/cynet/social/entity/BaseEntity.java
package com.cynet.social.entity;

import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import jdk.jfr.BooleanFlag;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public abstract class BaseEntity {

    @CreationTimestamp
    private LocalDateTime createdAt;
    @UpdateTimestamp
    private LocalDateTime updatedAt;
    @BooleanFlag
    private boolean isActive;
    @BooleanFlag
    private boolean isDeleted;
}
